src/GraphQLController.php

```diff
@@ -86,6 +98,61 @@ protected function queryContext(string $query, ?array $params, string $schema)
         }
     }
 
+    /**
+     * Note: it's expected this is called even when APQ is disabled to adhere
+     *       to the negotiation protocol.
+     *
+     * @param  string  $schemaName
+     * @param  array<string,mixed>  $input
+     * @return string
+     */
+    protected function handleAutomaticPersistQueries(string $schemaName, array $input): string
+    {
+        $query = $input['query'] ?? '';
+        $apqEnabled = config('graphql.apq.enable', false);
+
+        // Even if APQ is disabled, we keep this logic for the negotiation protocol
+        $persistedQuery = $input['extensions']['persistedQuery'] ?? null;
+        if ($persistedQuery && !$apqEnabled) {
+            throw AutomaticPersistedQueriesError::persistedQueriesNotSupported();
+        }
+
+        // APQ disabled? Nothing to be done
+        if (!$apqEnabled) {
+            return $query;
+        }
+
+        // No hash? Nothing to be done
+        $hash = $persistedQuery['sha256Hash'] ?? null;
+        if (null === $hash) {
+            return $query;
+        }
+
+        $apqCacheDriver = config('graphql.apq.cache_driver');
+        $apqCachePrefix = config('graphql.apq.cache_prefix');
+        $apqCacheIdentifier = "$apqCachePrefix:$schemaName:$hash";
+
+        $cache = cache();
+
+        // store in cache
+        if ($query) {
```

Comment:
Changed from `array_key_exists` on `$input` (and inverted the condition logic), I think it improves the readability here isn't technically strictly necessary in PHP